### PR TITLE
Removal of Identifiers

### PR DESCRIPTION
**Pull request by asparuhft**
# Description

## What

Removal of Identifier Nodes and Identifies relation

## Why

JIRA Ticket [UPPSF-1284](https://financialtimes.atlassian.net/browse/UPPSF-1284)

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [x] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

**Review by MiroslavGatsanoga — COMMENTED** [review]
The changes look good, can you deploy the release candidate on Dev so that we can test the behaviour.

**Comment by asparuhft** [comment]
The release candidate is deployed on dev now.

**Review by MiroslavGatsanoga — APPROVED** [review]

**Review by ivan-p-nikolov — DISMISSED** [review]

File: concepts/concepts_service.go [hunk 1]
```diff
@@ -653,9 +652,8 @@ func validateObject(aggConcept AggregatedConcept, transID string) error {
 		return requestError{formatError("sourceRepresentation", aggConcept.PrefUUID, transID)}
 	}
 	for _, concept := range aggConcept.SourceRepresentations {
-		// Is Authority recognised?
-		if _, ok := authorityToIdentifierLabelMap[concept.Authority]; !ok {
-			logger.WithTransactionID(transID).WithUUID(aggConcept.PrefUUID).Debugf("Unknown authority, therefore unable to add the relevant Identifier node: %s", concept.Authority)
+		if concept.Authority == "" {
```
Comment (epavlova) [inline concepts/concepts_service.go:655]: It's probably a good idea to still check that the `Authority` is not only non-empty string but is also one of the known authorities.

Comment (asparuhft) [reply concepts/concepts_service.go:655]: I considered doing it although the above logging was only at DEBUG level. One drawback is that we may miss some concepts if the list of supported authorities is not complete. In this case only four authorities were listed:
var authorityToIdentifierLabelMap = map[string]string{
	"TME":        "TMEIdentifier",
	"UPP":        "UPPIdentifier",
	"Smartlogic": "SmartlogicIdentifier",
	"FACTSET":    "FactsetIdentifier",
}

I have found the following list while looking at the public-concordances-api codebase:

// Map of authority to URI for the supported concordance IDs
var authorityMap = map[string]string{
	"TME":             "http://api.ft.com/system/FT-TME",
	"FACTSET":         "http://api.ft.com/system/FACTSET",
	"UPP":             "http://api.ft.com/system/UPP",
	"LEI":             "http://api.ft.com/system/LEI",
	"Smartlogic":      "http://api.ft.com/system/SMARTLOGIC",
	"ManagedLocation": "http://api.ft.com/system/MANAGEDLOCATION",
	"ISO-3166-1":      "http://api.ft.com/system/ISO-3166-1",
	"Geonames":        "http://api.ft.com/system/GEONAMES",
	"Wikidata":        "http://api.ft.com/system/WIKIDATA",
	"DBPedia":         "http://api.ft.com/system/DBPEDIA",
	"NAICS":           "http://api.ft.com/system/NAICS",
}

Which one is correct?

Comment (epavlova) [reply concepts/concepts_service.go:655]: The correct one is the second long list. Yes, you are correct that actually that was only debug logging, it was not returning error before. I've missed that. There are a few options: 
- add the authority check and keep it debug as was before;
- keep the PR as it is now;
- add the authority check and return error if the authority doesn't match our list (I would also check Neo for all possible authority values although the list above seems correct);
It's up to you. The third approach is actually changing the behaviour of the service so probably it's not a good idea to mess that together with the removal of the identifier nodes.

Comment (asparuhft) [reply concepts/concepts_service.go:655]: First option is implemented.

**Review by epavlova — APPROVED** [review]

**Review by ivan-p-nikolov — APPROVED** [review]

**Pull request merged**